packages/stream/solana/StreamClient.ts

```diff
@@ -170,6 +195,96 @@ export class SolanaStreamClient extends BaseStreamClient {
+    extParams: ICreateStreamSolanaExt,
+  ): Promise<{ ixs: TransactionInstruction[]; metadata: Keypair | undefined; metadataPubKey: PublicKey }> {
+    const {
+      tokenId: mint,
+      start,
+      period,
+      cliff,
+      canTopup,
+      cancelableBySender,
+      cancelableByRecipient,
+      transferableBySender,
+      transferableByRecipient,
+      partner,
+    } = streamParams;
+    const { isNative, sender, computeLimit, computePrice, metadataPubKeys, alignedConfigParams } = extParams;
+
+    if (!sender.publicKey) {
+      throw new Error("Sender's PublicKey is not available, check passed wallet adapter!");
+    }
+
+    const { minPrice, maxPercentage, minPercentage, maxPrice, oracleType, skipInitial, tickSize, priceOracle } =
+      alignedConfigParams!;
+
+    const { cliffAmount, amountPerPeriod, amount: depositedAmount } = recipient;
+
+    const recipientPublicKey = new PublicKey(recipient.recipient);
+    const mintPublicKey = isNative ? NATIVE_MINT : new PublicKey(mint);
+
+    const metadata = !metadataPubKeys ? Keypair.generate() : undefined;
+    const metadataPubKey = metadata ? metadata.publicKey : metadataPubKeys![0];
+
+    const { tokenProgramId } = await getMintAndProgram(this.connection, mintPublicKey);
+    const partnerPublicKey = partner ? new PublicKey(partner) : WITHDRAWOR_PUBLIC_KEY;
+
+    const streamflowProgramPublicKey = new PublicKey(this.programId);
+
+    const escrowPDA = deriveEscrowPDA(streamflowProgramPublicKey, metadataPubKey);
+
+    const oracle = priceOracle ?? new PublicKey("8an9aE6j4STjv1cNXaE7SJfqmfjgLUHpBKURjqcAQJbQ"); //deriveTestOraclePDA(this.program.programId, mintPublicKey, sender.publicKey!);
+
+    const ixs: TransactionInstruction[] = prepareBaseInstructions(this.connection, {
+      computePrice,
+      computeLimit,
+    });
+
+    const createIx = await this.alignedProxyProgram.methods
+      .create({
+        startTime: new BN(start),
+        netAmountDeposited: depositedAmount,
+        period: new BN(period),
+        amountPerPeriod: amountPerPeriod,
+        cliff: new BN(cliff),
+        cliffAmount: cliffAmount,
+        transferableBySender,
+        transferableByRecipient,
+        cancelableByRecipient,
+        cancelableBySender,
+        canTopup,
+        oracleType: oracleType ?? { switchboard: {} },
+        streamName: [],
+        minPrice,
+        maxPrice,
+        minPercentage,
+        maxPercentage,
+        tickSize: tickSize || new BN(1),
+        skipInitial: skipInitial ?? false,
+      })
+      .accountsPartial({
+        sender: sender.publicKey,
+        streamMetadata: metadataPubKey,
+        escrowTokens: escrowPDA,
+        mint: mintPublicKey,
+        partner: partnerPublicKey,
+        recipient: recipientPublicKey,
+        withdrawor: WITHDRAWOR_PUBLIC_KEY,
+        feeOracle: FEE_ORACLE_PUBLIC_KEY,
+        priceOracle: oracle,
+        tokenProgram: tokenProgramId,
+        streamflowProgram: this.programId,
```

Comment:
Some of these have fallbacks I use for testing, leaving them until I have tested it properly